### ISSUE TITLE
Can't use installed truffleruby 23.1.0 with openssl

### ISSUE THREAD

**Issue by HoneyryderChuck** [postmodern/ruby-install#469]
## Description

Coming from https://github.com/oracle/truffleruby/issues/3283

```
> ruby -e "require 'openssl'"


<internal:core> core/kernel.rb:260:in `require': you may need to install the system OpenSSL library libssl - see https://github.com/oracle/truffleruby/blob/master/doc/user/installing-libssl.md (dlopen(/user/.rubies/truffleruby-23.1.0/lib/mri/openssl.bundle, 0x0002): Library not loaded: @rpath/libgraalvm-llvm.1.dylib (RuntimeError)
  Referenced from: <D3AD165B-0AC8-3019-836C-9D3781CEDAA8> /user/.rubies/truffleruby-23.1.0/lib/mri/openssl.bundle
  Reason: tried: '/usr/local/opt/openssl@3/lib/libgraalvm-llvm.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/openssl@3/lib/libgraalvm-llvm.1.dylib' (no such file), '/usr/local/opt/openssl@3/lib/libgraalvm-llvm.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/openssl@3/lib/libgraalvm-llvm.1.dylib' (no such file), '/usr/local/lib/libgraalvm-llvm.1.dylib' (no such file), '/usr/lib/libgraalvm-llvm.1.dylib' (no such file, not in dyld cache))
        from /user/.rubies/truffleruby-23.1.0/lib/mri/openssl.rb:13:in `<top (required)>'
        from <internal:core> core/kernel.rb:260:in `require'
        from -e:1:in `<main>'
```

In the linked issue, it's mentioned that a certain post install hook isn't run in the process of installing truffleruby, which makes the openssl extension fail to link to libgraalvm.

## Environment

    $ ruby-install --version
    ruby-install: 0.9.2
    ...
    $ uname -a
    Darwin MacBook-Pro.lan 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; root:xnu-8796.141.3~6/RELEASE_X86_64 x86_64
    $ cc --version
    Apple clang version 15.0.0 (clang-1500.0.40.1)
    Target: x86_64-apple-darwin22.6.0
    Thread model: posix
    InstalledDir: /Library/Developer/CommandLineTools/usr/bin
    $ openssl version
    openssl version
    OpenSSL 1.1.1w  11 Sep 2023

**Comment by postmodern** [comment]
@HoneyryderChuck just to double check, does this happen after manually installing truffleruby? Is this a problem with `ruby-install`, or some platform issue with truffleruby?

**Comment by HoneyryderChuck** [comment]
According to the linked issue, ruby-install is not running a post install hook that it should. Can you confirm?

**Comment by postmodern** [comment]
@HoneyryderChuck `ruby-install` definitely runs the post-installation script:

* https://github.com/postmodern/ruby-install/blob/4971a1f934a687c1032e5e17615f791743e8c5d4/share/ruby-install/truffleruby/functions.sh#L66
* https://github.com/postmodern/ruby-install/blob/4971a1f934a687c1032e5e17615f791743e8c5d4/share/ruby-install/truffleruby-graalvm/functions.sh#L64-L92

Also, both `truffleruby` and `truffleruby-graalvm` has `openssl` as a dependency:

* https://github.com/postmodern/ruby-install/blob/4971a1f934a687c1032e5e17615f791743e8c5d4/share/ruby-install/truffleruby/dependencies.txt
* https://github.com/postmodern/ruby-install/blob/4971a1f934a687c1032e5e17615f791743e8c5d4/share/ruby-install/truffleruby-graalvm/dependencies.txt

On Fedora Linux I'm able to load `openssl` with `truffleruby-23.1.0` (`truffleruby`, haven't tested `truffleruby-graalvm`).

```shell
$ chruby truffleruby-23.1.0
$ ruby -e "require 'openssl'"
```

**Comment by postmodern** [comment]
@HoneyryderChuck how did you install `truffleruby` with `ruby-install`? `ruby-install truffleruby`? Also, I assume from the error message that your package manager is Homebrew?

**Comment by HoneyryderChuck** [comment]
yup, it's homebrew (I'm on Mac OS). I've done `ruby-install truffleruby 23.1.0`.

**Comment by eregon** [comment]
The problem seems not in ruby-install but in the macOS or XCode version used, so I think let's close this and continue on https://github.com/oracle/truffleruby/issues/3283

**Comment by todd-a-jacobs** [comment]
@postmodern There are multiple potential issues, one of which is that the latest target for ruby-install is 24.1.0 instead of 24.1.1, which builds under ruby-build. I haven't tried it manually with ruby-install, so it could just be a simple recipe change for some percentage of people. Please see below.

@HoneyryderChuck truffle-ruby 24.1.1 builds properly with ruby-build, but you have to restart your shell for chruby to see it under Fish (and possibly other shells) unlike rubies installed with ruby-install. Under the Fish shell, rubies installed with ruby-install are detected at the next prompt without having to re-initialize chruby state. For example:

```fish
~> ruby-build --version
ruby-build 20241030

~> ruby-build truffleruby-24.1.1 ~/.rubies/truffleruby-24.1.1

# Because macOS wants your Terminal shell to be a login shell. Plus,
# because of macOS-isms you may need to have SHELL set to zsh
# and then exec to a Homebrew-installed login shell from .zprofile
# or .zshrc even when your preferred shell is defined in /etc/shells.
#
# Custom shims not shown here, but various VNC and SSH clients
# don't work properly without Bash or Zsh as the current shell.
# YMMV, but chruby will still likely require you to exec a new login
# shell.
~> exec -- "$SHELL" --login

~> ruby -e "printf '%s-%s (%s)', RUBY_ENGINE, RUBY_ENGINE_VERSION, RUBY_VERSION"
truffleruby-24.1.1 (3.2.4)
```

The OP may also have other issues, but the primary one seems to be that openssl@1.1 has been removed from Homebrew, and TruffleRuby 24.1.0 doesn't seem to be able to build against openssl@3 in ruby-install. The upstream issue seems to be targeting a potentially different problem that isn't expected to be fixed until 24.1.2 in February, but if you need it *now* you may be able to either use ruby-build or update the chruby recipe or ruby-install options for TruffleRuby to target 24.1.1 instead of 24.1.0 depending on what the actual build issue really is for you.

**Comment by eregon** [comment]
I missed to push my changes to add 24.1.1 in ruby-versions, fixed now: https://github.com/postmodern/ruby-versions/commit/94adbd5766ebdd3e23f441407b6bab2b337e0f36

There should be no difference between 24.1.0 and 24.1.1 in terms of openssl though.

> The OP may also have other issues, but the primary one seems to be that openssl@1.1 has been removed from Homebrew, and TruffleRuby 24.1.0 doesn't seem to be able to build against openssl@3 in ruby-install.

24.1.0 builds fine against OpenSSL 1.1 and 3.
The OP issue should be solved with the resolution of https://github.com/oracle/truffleruby/issues/3283

>  The upstream issue seems to be targeting a potentially different problem that isn't expected to be fixed until 24.1.2 in February

Which issue are you thinking about? I don't see any issue linked to this one with a 24.1.2 milestone.
https://github.com/oracle/truffleruby/issues/3283 has been fixed in 23.1.2 almost a year ago.
